src/core/thread/mle.hpp

```diff
@@ -718,6 +722,27 @@ class Mle : public InstanceLocator, private NonCopyable
 
 #endif // OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
 
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+    /**
+     * Returns whether the Thread interface is currently communicating to a Wake-up Coordinator.
```

Comment:
I think in the context of MLE, we might use "WakeupParent" from the latest spec proposal to have the code better prepared for the scenario where `WC != WP`.